Metrics Configuration Produces Errors In Some Conditions

### Description

In the citus environment, the following errors were observed
```
2024-02-27T17:40:00.839Z  WARN pool-9-thread-1 c.g.b.c.c.BoundedLocalCache Exception thrown during refresh java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "com.hedera.mirror.importer.config.MetricsConfiguration$TableAttributes.partitioned()" because "table" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot invoke "com.hedera.mirror.importer.config.MetricsConfiguration$TableAttributes.partitioned()" because "table" is null
	at com.hedera.mirror.importer.config.MetricsConfiguration.getMetricSql(MetricsConfiguration.java:193)
	at com.hedera.mirror.importer.config.MetricsConfiguration.getUpdatedMetrics(MetricsConfiguration.java:151)
	at com.github.benmanes.caffeine.cache.CacheLoader.reload(CacheLoader.java:169)
	at com.github.benmanes.caffeine.cache.CacheLoader.lambda$asyncReload$2(CacheLoader.java:196)
	... 4 common frames omitted

2024-02-27T17:40:11.823Z  WARN pool-9-thread-1 c.g.b.c.c.BoundedLocalCache Exception thrown during refresh java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "com.hedera.mirror.importer.config.MetricsConfiguration$TableAttributes.partitioned()" because "table" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot invoke "com.hedera.mirror.importer.config.MetricsConfiguration$TableAttributes.partitioned()" because "table" is null
	at com.hedera.mirror.importer.config.MetricsConfiguration.getMetricSql(MetricsConfiguration.java:193)
	at com.hedera.mirror.importer.config.MetricsConfiguration.getUpdatedMetrics(MetricsConfiguration.java:151)
	at com.github.benmanes.caffeine.cache.CacheLoader.reload(CacheLoader.java:169)
	at com.github.benmanes.caffeine.cache.CacheLoader.lambda$asyncReload$2(CacheLoader.java:196)
	... 4 common frames omitted
```
This was caused by the names of some partitions of a table being named incorrectly. 

### Steps to reproduce

1. Change v2 schema so some partition names dont match parent.
2. Deploy and notice errors getting metrics

### Additional context

_No response_

### Hedera network

previewnet

### Version

v0.99.0

### Operating system

None

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/config/MetricsConfiguration.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/config/MetricsConfiguration.java
@@ -147,20 +147,20 @@ class MetricsConfiguration {
     }
 
     public TableMetrics getUpdatedMetrics(String tableName) {
-        var table = tables.get(tableName);
-        var sql = getMetricSql(table);
         try {
-            return jdbcOperations.queryForObject(
-                    sql, DataClassRowMapper.newInstance(TableMetrics.class), table.tableName());
+            var table = tables.get(tableName);
+            if (table != null) {
+                var sql = getMetricSql(table);
+                return jdbcOperations.queryForObject(
+                        sql, DataClassRowMapper.newInstance(TableMetrics.class), table.tableName());
+            }
         } catch (BadSqlGrammarException | EmptyResultDataAccessException e) {
-            // ignore as table may have been removed by a migration
+            // No longer need to query metrics for this table
+            tables.remove(tableName);
+            log.info("Removing {} and will stop querying metrics for this table", tableName);
         } catch (Exception e) {
-            log.warn("Error trying to get metrics for table {}", table.tableName(), e);
+            log.warn("Error trying to get metrics for table {}", tableName, e);
         }
-
-        log.info("Evicting {} from cache and will stop querying metrics for this table", tableName);
-        activeMetrics.invalidate(tableName);
-        tables.remove(tableName);
 
         return new TableMetrics(0L, 0L, 0L);
     }

